src/web/components/StickyBottomBanner/StickyBottomBanner.tsx

```diff
@@ -20,61 +24,75 @@ const getBannerLastClosedAt = (key: string): string | undefined => {
-    const [CMPWillShow, setCMPWillShow] = useState<boolean | undefined>(
-        undefined,
-    );
+    const [Banner, setBanner] = useState<React.FC | null>(null);
 
     useEffect(() => {
-        shouldShowOldCMP().then((shouldShowOld) =>
-            setShowOldCMP(shouldShowOld && CAPI.config.cmpUi),
-        );
-        willShowCMP().then(setCMPWillShow);
-    }, [CAPI.config.cmpUi]);
-
-    // Don't render anything until we know whether we can show the CMP
-    if (showOldCMP === null || CMPWillShow === null) {
-        return null;
-    }
+        if (isSignedIn === undefined) {
+            return;
+        }
 
-    // New CMP is not a react component and is shown outside of react's world
-    // so render nothing if it will show
-    if (CMPWillShow) return null;
+        const newCmp = {
+            id: 'cmpUi',
+            canShow: () =>
+                willShowCMP().then((result) => ({ result: !!result })),
+            // New CMP is not a react component and is shown outside of react's world
+            // so render nothing if it will show
+            show: () => null,
+            timeoutMillis: null,
+        };
+        const oldCmp = {
+            id: 'cmpUi',
+            canShow: () =>
+                shouldShowOldCMP().then((shouldShowOld) => ({
+                    result: shouldShowOld && CAPI.config.cmpUi,
+                })),
+            show: () => CMP,
+            timeoutMillis: null,
+        };
+        const readerRevenue = {
+            id: 'reader-revenue-banner',
+            canShow: () =>
+                canShowRRBanner({
+                    remoteBannerConfig: CAPI.config.remoteBanner,
+                    isSignedIn,
+                    asyncCountryCode,
+                    contentType: CAPI.contentType,
+                    sectionName: CAPI.sectionName,
+                    shouldHideReaderRevenue: CAPI.shouldHideReaderRevenue,
+                    isMinuteArticle: CAPI.pageType.isMinuteArticle,
+                    isPaidContent: CAPI.pageType.isPaidContent,
+                    isSensitive: CAPI.config.isSensitive,
+                    tags: CAPI.tags,
+                    contributionsServiceUrl: CAPI.contributionsServiceUrl,
+                    alreadyVisitedCount: getAlreadyVisitedCount(),
+                    engagementBannerLastClosedAt: getBannerLastClosedAt(
+                        'engagementBannerLastClosedAt',
+                    ),
+                    subscriptionBannerLastClosedAt: getBannerLastClosedAt(
+                        'subscriptionBannerLastClosedAt',
+                    ),
+                    switches: {
+                        remoteSubscriptionsBanner: !!CAPI.config
+                            .remoteSubscriptionsBanner,
+                    },
+                }),
+            /* eslint-disable-next-line react/jsx-props-no-spreading */
+            show: (meta: any) => () => <ReaderRevenueBanner {...meta} />,
```

Comment:
Note that by omitting the optional `timeoutMillis` parameter here we're automatically opting in the reader revenue banner to the 2 second timeout rule, which may not be what we want.